.github/workflows/ci.yml

```diff
@@ -0,0 +1,76 @@
+name: CI
+
+# Trigger the workflow on push or pull request, but only for the master branch
+on:
+  pull_request:
+  push:
+    branches: [master]
+
+jobs:
+  cabal:
+    name: ${{ matrix.os }} / ghc ${{ matrix.ghc }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - macOS-latest
+        cabal: ["latest"]
+        ghc:
+          - "8.4.4"
+          - "8.6.5"
+          - "8.8.4"
+          - "8.10.2"
+
+    steps:
+    - uses: actions/checkout@v2
+      #if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/master'
+
+    - uses: actions/setup-haskell@v1.1.4
+      id: setup-haskell-cabal
+      name: Setup Haskell
+      with:
+        ghc-version: ${{ matrix.ghc }}
+        cabal-version: ${{ matrix.cabal }}
+
+    - uses: actions/cache@v1
+      name: Cache cabal-store
+      with:
+        path: ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
+        key: ${{ runner.os }}-${{ matrix.ghc }}-cabal
+
+    - name: Build
+      run: |
+        cabal update
+        cabal build all --enable-tests --enable-benchmarks --write-ghc-environment-files=always
+    - name: Test
+      run: |
+        cabal test all --enable-tests
+  stack:
+    name: stack / ghc ${{ matrix.ghc }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        stack: ["latest"]
```

Comment:
Would also like to `stack build` against a few LTSes and OSes like we did with travis. Not immediately sure how to pair the `--resolver lts-##.yml` with a possibly necessary `--stack-yaml stack-lts-14.yml`...

Reply:
I added this as well with an additional matrix entry like the following:

```yaml
        resolver:
          - "--stack-yaml ./stack.yaml"
          - "--stack-yaml ./stack-lts-14.yaml"
          - "--resolver lts-15"
          - "--resolver lts-16"
```

This says that we'll have one build using our `./stack.yaml` (just to make sure our default stack setup works), one build using our `stack-lts-14.yaml` (to make sure the build works with lts-14), one build against the latest lts-15, and one build against the latest lts-16.

(The reason for doing `--stack-yaml ./stack-lts-14.yaml` and not `--resolver lts-14` is that we need an `extra-dep` that is not in lts-14).

Reply:
I took a shot at this in the last couple commits.